tests/Tests.AzureAppConfiguration/PushRefreshTests.cs

```diff
@@ -0,0 +1,313 @@
+            {
+              new PushNotification  {
+                                    ResourceUri = null,
+                                    EventType = "eventType.KeyValueModified",
+                                    SyncToken = "SyncToken1;sn=001"
+                                    },
+              new PushNotification  {
+                                    ResourceUri = new Uri("https://store2.resource.io/kv/searchQuery2"),
+
+									EventType = null,
+                                    SyncToken = "SyncToken2"
+                                    },
+              new PushNotification  {
+                                    ResourceUri = new Uri("https://store1.resource.io/kv/searchQuery1"),
+
+									EventType = "eventType.KeyValueDeleted",
+                                    SyncToken = null
+                                    },
+              new PushNotification  {
+                                    ResourceUri = null,
+                                    EventType = "eventType.KeyValueDeleted",
+                                    SyncToken = null
+                                    },
+              new PushNotification  {
+                                    ResourceUri = null,
+                                    EventType = null,
+                                    SyncToken = null
+                                    }
+            };
+
+		ConfigurationSetting FirstKeyValue => _kvCollection.First();
+
+        [Fact]
+        public void TryTryCreatePushNotification()
+        {
+            EventGridEvent eventGridEvent = new EventGridEvent()
+            {
+                Data = "{\"key\":\"searchQuery1\",\"etag\":\"etagValue1\",\"syncToken\":\"syncToken1;sn=001\"}",
+                DataVersion = "2",
+                EventTime = new DateTime(2021, 10, 6, 20, 08, 7),
+                EventType = "eventType.KeyValueModified",
+                Id = "some id",
+                Subject = "https://store1.resource.io/kv/searchQuery1",
+                Topic = "/subscriptions/subscription-value1/resourceGroups/resourceGroup1/providers/provider1/configurationstores/store1\\"
+            };
+
+			EventGridEvent badEventGridEventDataCapitalization = new EventGridEvent()
+			{
+				Data = "{\"Key\":\"searchQuery1\",\"Etag\":\"etagValue1\",\"SyncToken\":\"syncToken1;sn=001\"}",
+				DataVersion = "2",
+				EventTime = new DateTime(2021, 10, 6, 20, 08, 7),
+				EventType = "eventType.KeyValueModified",
+				Id = "some id",
+				Subject = "https://store1.resource.io/kv/searchQuery1",
+				Topic = "/subscriptions/subscription-value1/resourceGroups/resourceGroup1/providers/provider1/configurationstores/store1\\"
+			};
+
+			eventGridEvent.TryCreatePushNotification(out PushNotification pushNotification);
+
+			Assert.Equal(eventGridEvent.EventType, pushNotification.EventType);
+			Assert.Equal(new Uri(eventGridEvent.Subject), pushNotification.ResourceUri);
```

Comment:
>new Uri(eventGridEvent.Subject) [](http://example.com/codeflow?start=16&length=31)

You can use [Uri.OriginalString]( [https://docs.microsoft.com/en-us/dotnet/api/system.uri.originalstring?view=net-5.0](https://docs.microsoft.com/en-us/dotnet/api/system.uri.originalstring?view=net-5.0) ) to comapre the strings, instead of creating a new Uri object.